pkg/reconciler/taskrun/taskrun_test.go

```diff
@@ -4980,3 +4981,113 @@ status:
 		t.Errorf("expected Status.TaskSpec to match, but differed: %s", diff.PrintWantGot(d))
 	}
 }
+
+func TestReconcile_TestDefaultResults(t *testing.T) {
+
+	defaultStringResult := "default-result"
+	defaultArrayResult := []string{"default-result-a", "default-result-b"}
+	defaultObjectResult := map[string]string{"url": "foobar", "commit": "commit"}
+
+	defaultResultsTask := &v1beta1.Task{
+		ObjectMeta: objectMeta("test-results-task", "foo"),
+		Spec: v1beta1.TaskSpec{
+			Steps: []v1beta1.Step{simpleStep},
+			Results: []v1beta1.TaskResult{
+				{
+					Name:    "aResult",
+					Type:    v1beta1.ResultsTypeString,
+					Default: &v1beta1.ParamValue{StringVal: defaultStringResult},
+				},
+				{
+					Name:    "bResult",
+					Type:    v1beta1.ResultsTypeArray,
+					Default: &v1beta1.ParamValue{ArrayVal: defaultArrayResult},
+				},
+				{
+					Name:       "cResult",
+					Type:       v1beta1.ResultsTypeObject,
+					Properties: map[string]v1beta1.PropertySpec{"url": {Type: "string"}, "commit": {Type: "string"}},
+					Default:    &v1beta1.ParamValue{ObjectVal: defaultObjectResult},
+				},
+			},
+		},
+	}
+
+	taskRunWithResults := parse.MustParseTaskRun(t, `
+metadata:
+  name: test-taskrun-results-type-valid
+  namespace: foo
+spec:
+  taskRef:
+    name: test-results-task
+status:
+  taskResults:
+    - name: aResult
+      type: string
+      value: default-result
+    - name: bResult
+      type: array
+      value:
+      - default-result-a
+      - default-result-b
+    - name: cResult
+      type: object
+      value:
+        url: foobar
+        commit: commit
+`)
+
+	d := test.Data{
+		TaskRuns: []*v1beta1.TaskRun{taskRunWithResults},
+		Tasks:    []*v1beta1.Task{defaultResultsTask},
+		ConfigMaps: []*corev1.ConfigMap{{
+			ObjectMeta: metav1.ObjectMeta{Namespace: system.Namespace(), Name: config.GetFeatureFlagsConfigName()},
+			Data: map[string]string{
+				"enable-api-fields": config.AlphaAPIFields,
+			},
+		}},
+	}
+	for _, tc := range []struct {
+		name    string
+		taskRun *v1beta1.TaskRun
+	}{{
+		name:    "taskrun with default results",
+		taskRun: taskRunWithResults,
```

Comment:
we probably don't need to use table test if there's only one test case.
Or maybe we could add more test cases like,  if default is not set the taskrun should fail